### PR TITLE
Upgrade export map

### PR DESCRIPTION
**Pull request by tltk90**

### CODE REVIEW

File: src/app/@ansyn/ansyn/modules/menu-items/tools/export-maps-popup/export-maps-popup.component.html [hunk 1]
```diff
@@ -1,24 +1,70 @@
...
+<mat-dialog-content>
+	<h5 mat-dialog-title>{{description}}</h5>
+	<div class="method-chose">
+		<ansyn-radio
+			[(ngModel)]="exportMethod"
+			[value]="basicExport"
+			name="basicExport">
+			{{basicExport | translate}}
+		</ansyn-radio>
+		<ansyn-radio
```
Comment (sulimani55) [inline src/app/@ansyn/ansyn/modules/menu-items/tools/export-maps-popup/export-maps-popup.component.html:14]: maybe create a config for the basic and advanced exports, and use ngFor

File: src/app/@ansyn/ansyn/modules/menu-items/tools/export-maps-popup/export-maps-popup.component.ts [hunk 2]
```diff
@@ -1,99 +1,304 @@
...
+import { IToolsConfig, toolsConfig } from '../models/tools-config';
+import { annotationsClassNameForExport } from '@ansyn/ol';
+import { measuresClassNameForExport } from '../../../plugins/openlayers/plugins/visualizers/tools/measure-distance.visualizer';
+import { IExportMapData, IExportMapMetadata, ImageryCommunicatorService, toDegrees } from '@ansyn/imagery';
+import { jsPDF } from "jspdf";
+import { TranslateService } from '@ngx-translate/core';
+import { IOverlay } from '../../../overlays/models/overlay.model';
+import { Observable, of, EMPTY } from 'rxjs';
 
+enum GraphicExportEnum {
```
Comment (sulimani55) [inline src/app/@ansyn/ansyn/modules/menu-items/tools/export-maps-popup/export-maps-popup.component.ts:25]: i think moving this to a seperate config file would be more right

File: src/app/@ansyn/ansyn/modules/menu-items/tools/export-maps-popup/export-maps-popup.component.ts [hunk 3]
```diff
@@ -1,99 +1,304 @@
...
+	constructor(protected store$: Store<any>,
+				protected logger: LoggerService,
+				protected dialogRef: MatDialogRef<ExportMapsPopupComponent>,
+				protected imageryCommunicatorService: ImageryCommunicatorService,
+				protected translateService: TranslateService,
+				@Inject(DOCUMENT) protected document: any,
+				@Inject(toolsConfig) public toolsConfigData: IToolsConfig) {
 		this.logger.info(LOGS.request);
-		this.store$.dispatch(new SetMinimalistViewModeAction(true));
+		this.translateService.get('direction', '').subscribe( (direction) => this.direction = direction === 'rtl' ? 'rtl' : 'ltr')
```
Comment (sulimani55) [inline src/app/@ansyn/ansyn/modules/menu-items/tools/export-maps-popup/export-maps-popup.component.ts:136]: is the translateService unsubscribes by default?

File: src/app/@ansyn/ansyn/modules/menu-items/tools/export-maps-popup/export-maps-popup.component.ts [hunk 4]
```diff
@@ -1,99 +1,304 @@
...
-			}).catch( err => {
+
+				doc.save('map.pdf');
+				this.logger.info(LOGS.success);
+			}),
+			catchError( (err) => {
+				console.error(err);
 				this.logger.error(LOGS.failed);
-				this.closeModal();
+				this.store$.dispatch(new SetToastMessageAction({toastText: `can't export map,use basic export instead`}));
```
Comment (sulimani55) [inline src/app/@ansyn/ansyn/modules/menu-items/tools/export-maps-popup/export-maps-popup.component.ts:202]: add the text 'can't export map,use basic export instead' to the toastMessages object

File: src/app/@ansyn/ansyn/modules/menu-items/tools/export-maps-popup/export-maps-popup.component.ts [hunk 5]
```diff
@@ -1,99 +1,304 @@
...
+		if (!mapToBeExport) {
+			throw new Error('No Such Map Error');
+		}
+
+
+		mapToBeExport.exportMap(exportMetadata).pipe(
+			mergeMap( (exportMapData) => {
+				if (exportMetadata.extra.north) {
+					return new Observable( obs => {
+						const rotation = mapToBeExport.getRotation();
```
Comment (sulimani55) [inline src/app/@ansyn/ansyn/modules/menu-items/tools/export-maps-popup/export-maps-popup.component.ts:157]: can you elaborate on the function?

File: src/app/@ansyn/imagery/communicator-service/communicator.entity.ts [hunk 6]
```diff
@@ -347,4 +357,16 @@ export class CommunicatorEntity implements OnInit, OnDestroy {
 		}
 	}
 
+	private calcBoundingRect() {
+		const boundingRect = this.ActiveMap.getHtmlContainer().getBoundingClientRect();
+		// cut the status bar and the menu
+		return {
+			x: Math.round(boundingRect.left - 90),
```
Comment (sulimani55) [inline src/app/@ansyn/imagery/communicator-service/communicator.entity.ts:364]: what do 90 and 55 mean

File: src/app/@ansyn/ol/maps/helpers/helpers.ts [hunk 7]
```diff
@@ -0,0 +1,43 @@
+import { Observable } from 'rxjs';
+
+export function exportMapHelper(mapObject, size, resolution, classToInclude): Observable<HTMLCanvasElement> {
+	const width = Math.round((size[0] * resolution) / 25.4);
```
Comment (sulimani55) [inline src/app/@ansyn/ol/maps/helpers/helpers.ts:4]: can you elaborate on this function?
and what does 'helper' mean in the context of the project

**Review by sulimani55 — CHANGES_REQUESTED** [review]

File: src/app/@ansyn/imagery/communicator-service/communicator.entity.ts (hunk 6 above)
Comment (tltk90) [reply src/app/@ansyn/imagery/communicator-service/communicator.entity.ts:364]: // cut the status bar and the menu

File: src/app/@ansyn/ol/maps/helpers/helpers.ts (hunk 7 above)
Comment (tltk90) [reply src/app/@ansyn/ol/maps/helpers/helpers.ts:4]: it contain function that the help the ol module.
export map, is the same for `ol-map` and `disable-ol-map`

File: src/app/@ansyn/ansyn/modules/menu-items/tools/export-maps-popup/export-maps-popup.component.ts (hunk 5 above)
Comment (tltk90) [reply src/app/@ansyn/ansyn/modules/menu-items/tools/export-maps-popup/export-maps-popup.component.ts:157]: on the advance export we use the map object.
for export in better resolution,
so the compass was not there as it not part of the map.
so if the user select to export it i add it manually.

File: src/app/@ansyn/imagery/communicator-service/communicator.entity.ts (hunk 6 above)
Comment (sulimani55) [reply src/app/@ansyn/imagery/communicator-service/communicator.entity.ts:364]: are those represent the sizes of the status bar and the menu?

File: src/app/@ansyn/ansyn/modules/menu-items/tools/export-maps-popup/export-maps-popup.component.ts (hunk 3 above)
Comment (tltk90) [reply src/app/@ansyn/ansyn/modules/menu-items/tools/export-maps-popup/export-maps-popup.component.ts]: yes.

File: src/app/@ansyn/ansyn/modules/menu-items/tools/export-maps-popup/export-maps-popup.component.ts (hunk 2 above)
Comment (tltk90) [reply src/app/@ansyn/ansyn/modules/menu-items/tools/export-maps-popup/export-maps-popup.component.ts:25]: i don't agree. it is not relevant to any file except this one.

Comment (sulimani55) [reply src/app/@ansyn/ansyn/modules/menu-items/tools/export-maps-popup/export-maps-popup.component.ts:25]: so don't you think it should have config of its own?
it makes the file less overwhelming

File: src/app/@ansyn/ansyn/modules/menu-items/tools/export-maps-popup/export-maps-popup.component.html (hunk 1 above)
Comment (tltk90) [reply src/app/@ansyn/ansyn/modules/menu-items/tools/export-maps-popup/export-maps-popup.component.html]: 👍

**Review by sulimani55 — APPROVED** [review]

**Pull request merged**